src/gsi-ncdiag/combine_conv.py.in

```diff
@@ -154,13 +156,126 @@ def concat_ioda(FileList, OutFile):
             for i in range(len(tmp)):
                 for j in range(len(tmp[i])):
                     tmp2[i, j] = str(tmp[i][j])
-            loc_mdata[vname[0]] = tmp2.astype(MetaVType[idx])
+            loc_mdata[vname[0]] = tmp2.astype(MetaVType[idx3])
         else:
-            loc_mdata[vname[0]] = MetaVarUnique[idx, ...].astype(MetaVType[idx])
-    for idx, vname in enumerate(DataVars):
-        outdata[tuple(vname.split('@'))] = DataVarUnique[idx, ...].astype(DataVType[idx])
+            loc_mdata[vname[0]] = MetaVarUnique[idx3, ...].astype(MetaVType[idx3])
+    for idx3, vname in enumerate(DataVars):
+        tmp = DataVarUnique[idx3, ...]
+        tmp = tmp.astype(DataVType[idx3])
+        if DataVType[idx3] == 'int32':
+            tmp[tmp < -1e5] = nc.default_fillvals['i4']
+        outdata[tuple(vname.split('@'))] = tmp
     writer.BuildNetcdf(outdata, rec_mdata, loc_mdata, var_mdata, AttrData)
 
+    # now write out combined GeoVaLs file
+    if GeoDir:
+        # get list of geoval files
+        GeoFileList = []
+        GeoVarNames2 = []
+        GeoVarTypes2 = []
+        GeoVarNames3 = []
+        GeoVarTypes3 = []
+        for f in FileList:
+            inob = f.split('/')[-1]
+            ingeo = inob.replace('obs', 'geoval')
+            g = GeoDir+'/'+ingeo
+            GeoFileList.append(g)
+        # figure out possible variable names
+        for f in GeoFileList:
+            ncf = nc.Dataset(f, mode='r')
+            for key, value in ncf.variables.items():
+                if key not in GeoVarNames2 and key not in GeoVarNames3:
+                    if value.ndim == 1:
+                        GeoVarNames2.append(key)
+                        GeoVarTypes2.append(ncf.variables[key].dtype)
+                    else:
+                        GeoVarNames3.append(key)
+                        GeoVarTypes3.append(ncf.variables[key].dtype)
+            ncf.close()
+        GeoVarData2 = []
+        GeoVarIdx2 = []
+        GeoVarData3 = []
+        GeoVarIdx3 = []
+        for idx2, v in enumerate(GeoVarNames2):
+            tmpgeodata = []
+            tmpgeoidx = []
+            for f in GeoFileList:
+                ncf = nc.Dataset(f, mode='r')
+                try:
+                    tmpdata = ncf.variables[v][:]
+                    tmpgeodata.append(tmpdata)
+                    tmpgeoidx.append(np.ones_like(tmpdata).astype(int)*int(idx2))
+                except KeyError:
+                    tmpdata = np.ones_like(ncf.variables['time'][:]).astype(GeoVarTypes2[idx2])
+                    if GeoVarTypes2[idx2] == np.int32:
+                        tmpdata = tmpdata * nc.default_fillvals['i4']
+                    else:
+                        tmpdata = tmpdata * np.abs(nc.default_fillvals['f4'])
+                    tmpgeodata.append(tmpdata)
```

Comment:
The default fill value for integers is negative and for floats is positive. The np.abs() was originally applied to make sure a large positive number went to IODA since, at that time, the IODA reader only recognized a large positive number as a missing value. Now the IODA reader recognizes the actual netcdf missing values, so the np.abs() call isn't necessary anymore.

If there is an integer using the absolute value of the integer fill value, then IODA will still issue a warning since it is expecting the actual default fill value which is negative. 

The absolute value of the floating point fill value doesn't hurt anything since the actual fill value is positive.

The np.abs() calls aren't necessary now and probably should be removed at some point.

Reply:
Gotcha, I went ahead and updated the code to remove the np.abs calls and updated the radiance test file to get all of the ctests to pass again (it had 0 and not a negative large value for missing PreQC values)